actionpack/lib/action_controller/metal/mime_responds.rb

```diff
@@ -195,10 +195,14 @@ def respond_to(*mimes)
       yield collector if block_given?
 
       if format = collector.negotiate_format(request)
-        _process_format(format)
-        _set_rendered_content_type format
-        response = collector.response
-        response.call if response
+        if self.content_type && self.content_type != format.to_s
```

Comment:
You shouldn't need to use `self` here. Also if we're banning nested `respond_to` calls we can just do something like `raise AbstractController::DoubleRespondToError if content_type` right?